### ISSUE TITLE
app disappears

### ISSUE THREAD

**Issue by debs289** [sphinx02/FireStarter#51]
I installed firestarter and kodi following your instructions but firestarter keeps disappearing off manage installed apps? What am i doing wrong

**Comment by JazzyJeffaz** [comment]
You aren't doing anything wrong. Amazon pushed an update that makes it so Firestarter won't work.

**Comment by jkchr1s** [comment]
Yep, @JazzyJeffaz is correct. It appears they are blacklisting the package by name. It is automatically disabled and hidden by default, and without root, you don't have permission to re-enable it.

**Comment by debs289** [comment]
So can we not put it on then

**Comment by datacrashfixer4u** [comment]
Is there another option to install Firestarter. Amazon doesnt realize that people who use Firestarter also use it to upgrade Kodi and to be able to boot apps that the user paid Amazon for being able to Utilize their Fire TV's and Firesticks.

**Comment by jkchr1s** [comment]
See #52 changing the package name fixed it for me.

**Comment by dazza1414** [comment]
how do we change the package name to enable firetarter to work?

**Comment by jkchr1s** [comment]
You can refactor the src/main/java/de/belu/firestarter directory in android studio then recompile. You'll also need to bundle an executable of adb. I have already done this in my fork https://github.com/jkchr1s/FireStarter and sent a PR ( see #52 ) .

**Comment by bigpearl28** [comment]
@jkchr1s how do I download an appointment of your fork

**Comment by debs289** [comment]
I saw a post on another site about blocking amazon updates...would this work?

**Comment by meliton** [comment]
@jkchr1s is right. I just did the same thing under a different name and it now survives reboots.
I did it the long way using Notepad++ Replace in Files feature to hunt down de.belu.firestarter and renamed a few of the firestarter files to match the new file name. Then recompiled in android studio. FYI, FiredTV is also disappearing

**Comment by infinityhaitian** [comment]
What is the next step to make the firestarter working again?

**Comment by delaware1999** [comment]
Hello Sphinx02   can you please tell us all if there is a new solution comming ? Yes or NO ?
Sincd 3 Days you didnt write anything about this case that Firestarter is not working..

**Comment by datacrashfixer4u** [comment]
delaware1999,  apparently you think that sphin02 owes you something,  developers of Android apps receive nothing for their contributions all they ask is that they receive some sort of acknowledgement for their contributions and the occasional cup of coffee. So if you think that you have a right for using their code,  then you try reverse-engineering the code or instead be thankful developers like sphinx02 give a damm to try to solve this issue. 
On Apr 5, 2016 11:19 PM, delaware1999 notifications@github.com wrote:Hello Sphinx02   can you please tell us all if there is a new solution comming ? Yes or NO ?
Sincd 3 Days you didnt write anything about this case that Firestarter is not working..

**Comment by delaware1999** [comment]
And datacrashfixer4u ? did I ask you something ? So why do you answer ? I dont know you and dont want to talk to you. I asked a simple question and thats what other thousends users are asking and dont not write it because they are afraid !! What is your problem ?
Off course he doesnt get any money for it ! I Understand. But a little massage would be no problem !

**Comment by delaware1999** [comment]
I found a solution. I got the Firestopper. it works now with 5.0.5.1 Version..

**Comment by datacrashfixer4u** [comment]
I am sorry if I offended you,  whereas I realise that you may be afraid
that you will not be able to continue the joy of not having to pay for
bundled crap satellite or cable TV,  remember that sphinx02 and his fellow
deveopers will suffer as well. But remember that you will get more results
with a spoon of honey than a spoon of vinegar. One more thing,  remember
that it is better for the Firestarter folks to design a cure for Amazon's
ignorance than simply a bandaid. Hope you understand?

**Comment by infinityhaitian** [comment]
i just want to say thank you for firestater. I do appreciate every effort to make this work for all of us.

**Comment by dazza1414** [comment]
So how do I get firestarter to work again on my stick?

Daryl Jenkins

On 6 Apr 2016, at 1:01 pm, delaware1999 <notifications@github.com<mailto:notifications@github.com>> wrote:

I found a solution. I got the Firestopper. it works now with 5.0.5.1 Version..

##

**Comment by delaware1999** [comment]
You must find the Firestopper.exe  give me a mail and I send you a link

**Comment by infinityhaitian** [comment]
jearuiz0@gmail.com

**Comment by delaware1999** [comment]
mail is send. remove your mail !

**Comment by infinityhaitian** [comment]
how

**Comment by infinityhaitian** [comment]
thank you

**Comment by infinityhaitian** [comment]
how do i get the firestarter to work again?

**Comment by delaware1999** [comment]
remove the old version !  Install this one I send you via ADB Sideload  and then you got it.. I cant help you with that. There are many instructions in the Internet. use Google Search please how to install a APK into your Fire TV

**Comment by infinityhaitian** [comment]
I already know the instructuions

thanks again

**Comment by infinityhaitian** [comment]
thanks again

**Comment by dazza1414** [comment]
Could you send me the link pls too?

dazjenkins@hotmail.commailto:dazjenkins@hotmail.com

Daryl Jenkins

On 6 Apr 2016, at 1:38 pm, delaware1999 <notifications@github.com<mailto:notifications@github.com>> wrote:

You must find the Firestopper.exe give me a mail and I send you a link

##

**Comment by jayeshjethwa** [comment]
Hi Delaware 

can you send me the link too @delaware1999

**Comment by mknasty** [comment]
Delaware, can i get the link too? mknasty@aol.com ....... thanks!!!!

**Comment by datacrashfixer4u** [comment]
OK delaware1999 could you please send me the link as well. We know this is just a bandaid but hopefully the developers will come up with a real good fix. My mail is datacrashfixer4u@comcast.net. Thsnks.

**Comment by jkchr1s** [comment]
It's in the XDA thread...

**Comment by wpinfos** [comment]
What do you mean with XDA thread - i failed to find it :(

**Comment by jkchr1s** [comment]
http://forum.xda-developers.com/showpost.php?p=66215906&postcount=592

**Comment by decantbob** [comment]
my firestarter quit working too could you please email me the new link for fire tv  .      thank you my email   decantbob@yahoo.com

**Comment by jesterbw** [comment]
Hello Delaware
can you please send me the link s well.
jesterbw@hotmail.com
Thank you

**Comment by rem009** [comment]
Can i please get a link for this as well coolascanbe45@gmail.com

**Comment by jkchr1s** [comment]
There's no need to get links via email. There is no "firestarter.exe" and if you downloaded one, I would suggest you get rid of it. Android apps are not .exe files, they are .apk files. Use the xda-developers link posted above to download it.

**Comment by rem009** [comment]
Thank you very much. I just found it. 

Ms. Mendoza

**Comment by GainfulShrimp** [comment]
I had this problem, but there's a simple solution - for easily accessing Kodi anyway - without installing anything:
Home Screen > Apps > Your Apps Library > Kodi > Add to Favourites
Then, Kodi appears on the home screen in the Recent section... easy peasy!
I'm pretty sure that this didn't work in older versions of Fire TV software, so although Amazon have blocked Firestarter, they now apparently allow you to add sideloaded apps to the regular home screen, which seems a neater solution than using Llama or Firestarter etc anyway... 
It's still a bummer for anyone who was using some feature of Firestarter that I don't use, but for easily accessing Kodi (all I personally used Firestarter for), having it right there in Recent apps works for me. :)

**Comment by dazza1414** [comment]
But what about updating kodi? That's what I used to use firestarter for aswell as configuring the double tap for kodi

Daryl Jenkins

On 7 Apr 2016, at 10:19 pm, GainfulShrimp <notifications@github.com<mailto:notifications@github.com>> wrote:

I had this problem, but there's a simple solution - for easily accessing Kodi anyway - without installing anything:
Home Screen > Apps > Your Apps Library > Kodi > Add to Favourites
Then, Kodi appears on the home screen in the Recent section... easy peasy!
I'm pretty sure that this didn't work in older versions of Fire TV software, so although Amazon have blocked Firestarter, they now apparently allow you to add sideloaded apps to the regular home screen, which seems a neater solution than using Llama or Firestarter etc anyway...
It's still a bummer for anyone who was using some feature of Firestarter that I don't use, but for easily accessing Kodi (all I personally used Firestarter for), having it right there in Recent apps works for me. :)

##

**Comment by datacrashfixer4u** [comment]
As was previously stated, Firestarter has built-in utilities that allow us the  customers who bought Amazon's product to use Kodi, hell with them,  I have found numerous devices on the Internet that will not be stifled by Amazon imature intent on choking what is the future of television. 
On Apr 7, 2016 3:38 PM, dazza1414 notifications@github.com wrote:But what about updating kodi? That's what I used to use firestarter for aswell as configuring the double tap for kodi

Daryl Jenkins

On 7 Apr 2016, at 10:19 pm, GainfulShrimp <notifications@github.com<mailto:notifications@github.com>> wrote:

I had this problem, but there's a simple solution - for easily accessing Kodi anyway - without installing anything:
Home Screen > Apps > Your Apps Library > Kodi > Add to Favourites
Then, Kodi appears on the home screen in the Recent section... easy peasy!
I'm pretty sure that this didn't work in older versions of Fire TV software, so although Amazon have blocked Firestarter, they now apparently allow you to add sideloaded apps to the regular home screen, which seems a neater solution than using Llama or Firestarter etc anyway...
It's still a bummer for anyone who was using some feature of Firestarter that I don't use, but for easily accessing Kodi (all I personally used Firestarter for), having it right there in Recent apps works for me. :)

##

**Comment by baddog814** [comment]
Hay just a thought but don't we own the hardware, and therefor have the right to install what we want on the stick.

**Comment by infinityhaitian** [comment]
yes, you are right about that.. we should have the right to do whatever we like with it

**Comment by datacrashfixer4u** [comment]
Not really so it seems. Then I guess Google should get involved as Amazon is reverse-engineering FireOS to deprive we the people that put out our hard-earned money to buy their FireTV's and Firesticks from running Kodi or utilities that help us maintain Kodi. 
On Apr 7, 2016 4:20 PM, baddog814 notifications@github.com wrote:Hay just a thought but don't we own the hardware, and therefor have the right to install what we want on the stick.

**Comment by meliton** [comment]
@infinityhaitian and @baddog814, I COMPLETELY agree with you. Now here's the bad news... Amazon's Conditions of Use and Terms of Service state you are only **'licensed'** to use their software and "_All software used in any Amazon Service is the property of Amazon or its software suppliers._" It's similar to Apple's stranglehold on their hardware and software supply chain.

**Comment by baddog814** [comment]
Yes but Microsoft has the same agreement but they have never interfered with any hardware and if they have that problem the hardware and the software are not the same

**Comment by baddog814** [comment]
Furthermore once you own the hardware do you really need their software on it at all maybe we should just delete it

**Comment by datacrashfixer4u** [comment]
Guess it is up to Google to look into this. They own Android and I am sure when they got a license to use Android in their Fire OS which is Android there is a clause in their license about them preventing Android software not being utilized on their platform. 
On Apr 7, 2016 5:35 PM, Mel notifications@github.com wrote:@infinityhaitian and @baddog814, I COMPLETELY agree with you. Now here's the bad news... Amazon's Conditions of Use and Terms of Service state you are only 'licensed' to use their software and "All software used in any Amazon Service is the property of Amazon or its software suppliers." It's similar to Apple's stranglehold on their hardware and software supply chain.

**Comment by datacrashfixer4u** [comment]
Why not,  but how do we reprogram their rom in the Fire TV or Firesticks? I am sur there is a way. As my professor said in college; where there is a will there is a way. 
On Apr 7, 2016 6:33 PM, baddog814 notifications@github.com wrote:Furthermore once you own the hardware do you really need their software on it at all maybe we should just delete it

**Comment by ghost** [comment]
Can someone please make a "Kodi Updater" app?  This would be much simpler than FireStarter.  Maybe amazon wouldn't ban "Kodi Updater".  Maybe they ban FireStarter because they don't like it taking over the home button.  If they do ban "Kodi Updater" explicitly then at least we know where they stand.

**Comment by baddog814** [comment]
That's the same agreement that you make with Microsoft Windows you only
lease their software you never owned

Paul

**Comment by dap0927** [comment]
Hello to all my fellow bloggers.

Can anyone tell me where can I download a copy of the open source for firestarter. I want to learn how to create apps like this one in order to better support folks like us with our interaction with android and Kodi.

Also what tools (compilers, books etc) I need for this. My knowledge is very basic at the monent.

Thanks

DarcMan

**Comment by ghost** [comment]
DarcMan:  I wish you success but you're gonna have to get the hang of Google and be a bit more resourceful in order to achieve your goals.  Here's a hint:  You are commenting on an issue tracking system for where the source is being hosted.  :)  First, as baby steps, maybe learn what git is, and what github is.

**Comment by dap0927** [comment]
.....thank you....I think. I know the source code is available somewhere in the github site I just couldn't find it as the dite format doesn't seem too user friendly to me but thanks.

**Comment by meliton** [comment]
@jkchr1s has done some great work on updating FireStarter and refactoring it to Firestopper. Since his code is now public, Amazon will probably cat and mouse Firestopper with an update. The good news is that you can use his code and refactor it into another package and you'll be immune from a future blacklisted Firestopper. If you do that, be sure not to publicly announce your refactored package since Amazon can see it and could easily blacklist YOUR package. I've already got a few non-public apks going and they are running under 5.0.5.1. Thank you open source!

**Comment by dccxxvi** [comment]
So when amazon updated their fireos it removed both kodi and firestarter, maybe because it was linkewd together somehow. 

I had to reconfigure kodi to my settings which is a PITA. Can I install firestopper, update kodi, uninstall firestopper and have kodi safe from a "firestarter/firestopper" removal?

**Comment by meliton** [comment]
@dccxxvi, the update didn't kill Kodi... it killed Firestarter, and soon Firestopper will be next. You should get a refactored copy of Firestopper that installs in an "Unknown to Amazon" location and you'll be all set.

**Comment by wpinfos** [comment]
How can i refactor something - i have no clue and would appreciate any howto :)
Thanks

**Comment by dccxxvi** [comment]
@meliton when my FTV updated sunday it actually removed both! So far I am still good.

**Comment by datacrashfixer4u** [comment]
Are you saying that Amazon's upgrade on sunday removed both your firestopped and Kodi when hou did your Amazon upgrade. 
On Apr 12, 2016 6:48 AM, dccxxvi notifications@github.com wrote:@meliton when my FTV updated sunday it actually removed both! So far I am still good.

**Comment by datacrashfixer4u** [comment]
I haven't programmed in years but I would also like assistance to refactor as I have never programmed in Android? 
On Apr 12, 2016 12:43 AM, wpreview notifications@github.com wrote:How can i refactor something - i have no clue and would appreciate any howto :)
Thanks

**Comment by dazza1414** [comment]
Firstly could somebody please explain to me in the most Lehman terms if and how I can get firestarter to work again on my stick?
Secondly if not is there an alternative way to update kodi on my stick as I used to use the firestarter app for this?

Daryl Jenkins

On 12 Apr 2016, at 4:50 pm, datacrashfixer4u <notifications@github.com<mailto:notifications@github.com>> wrote:

I haven't programmed in years but I would also like assistance to refactor as I have never programmed in Android?
On Apr 12, 2016 12:43 AM, wpreview <notifications@github.com<mailto:notifications@github.com>> wrote:How can i refactor something - i have no clue and would appreciate any howto :)
Thanks

-You are receiving this because you commented.Reply to this email directly or view it on GitHub

##

**Comment by decantbob** [comment]
Go to apps on home page and go to ES go to top of page on left to favorite click go to add and type in 
http://Kodi.tv/download

Bob DeCant
419-944-0006
Massive Action
Massive Results

**Comment by meliton** [comment]
@wpreview @datacrashfixer4u Here's a quick tutorial on recompiling Firestopper. [Recompiling FireStopper Tutorial](http://www.filedropper.com/recompilingfirestoppertutorial_1). If there's a better way, let me know, but this worked for me.

**Comment by datacrashfixer4u** [comment]
I do not have an amazon account,  is there somewhere else besides
filestopper where I may download the tutorial?

**Comment by meliton** [comment]
The link should open up filedropper...
http://www.filedropper.com/recompilingfirestoppertutorial_1

**Comment by infinityhaitian** [comment]
I noticed that the home button double click does not work anymore, even after I selected firestopper for for the home button double click, it still did not work. please help

**Comment by delaware1999** [comment]
13 days now passed.. Is there a new Version comming from the Coder ? 
Or did he get a Mail from Amazon and they suit him now ?  Can you please inform us if you will release ever a new version that works.

**Comment by datacrashfixer4u** [comment]
Tonight when I went to use my firestick at about 11:30 pacific time,  I discovered that my firestick had to be reconfigured for the logon on to my network and also discovered that my firestopper program had been deleted and that my Kodi was not visible on the home menu and could only be accessed through settings. It was able to be launched. Has anyone out there experienced the same problem? 
On Apr 15, 2016 2:52 AM, delaware1999 notifications@github.com wrote:13 days now passed.. Is there a new Version comming from the Coder ? 
Or did he get a Mail from Amazon and they suit him now ?  Can you please inform us if you will release ever a new version that works.

**Comment by meliton** [comment]
@datacrashfixer4u , is your Firestick firmware at 5.0.5.1 or did it change?

**Comment by datacrashfixer4u** [comment]
It is 5.0.5.1(537175520)
On Apr 21, 2016 3:09 AM, Mel notifications@github.com wrote:@datacrashfixer4u , is your Firestick firmware at 5.0.5.1 or did it change?

**Comment by datacrashfixer4u** [comment]
Apparently Amazon is able to push through updates if you leave your
firestick powered on even if not using its input to your television.
On Apr 21, 2016 3:09 AM, "Mel" notifications@github.com wrote:

@datacrashfixer4u https://github.com/datacrashfixer4u , is your Firestick
firmware at 5.0.5.1 or did it change?

**Comment by meliton** [comment]
If your Firestick is powered, it is still connected to your wireless Internet so Amazon can still push updates that kill Firestarter, Firestopper and whatever else they've identified. I've been using a custom Firestopper since this whole debacle happened and I haven't had any issues. Amazon can kill what they know, so refactoring the program is the way to go.

**Comment by ricky2389** [comment]
Just a thought, i'm not a programmer or anything but from doing the workaround for firestarter, i noticed a load of app names from amazon. To block future updates, is it not a case of simply uninstalling the update service? I dont know what its called but wondered if it was that simple??

**Comment by ghost** [comment]
ricky2389..  Apparently.   http://www.aftvnews.com/how-to-block-software-updates-on-the-amazon-fire-tv-or-fire-tv-stick/#withoutroot

**Comment by castalla** [comment]
quote - so refactoring the program is the way to go.- unquote

Any guide/tutorial on how to ?

**Comment by meliton** [comment]
I posted tutorials in this thread about 10 posts ago. It's worked perfectly for my Fire sticks and Fire TVs.
http://www.filedropper.com/recompilingfirestoppertutorial_1

**Comment by decantbob** [comment]
I do not have that post could you resend it to me thank you

Bob DeCant
419-944-0006
Massive Action
Massive Results

**Comment by meliton** [comment]
Click this link, then click "Download this File", then enter the CAPTCHA...
http://www.filedropper.com/recompilingfirestoppertutorial_1

**Comment by castalla** [comment]
Thanks.
